include/swift/AST/DiagnosticsSema.def

```diff
@@ -7870,7 +7867,7 @@ ERROR(noncopyable_types_cannot_be_resilient, none,
       "supported",
       (const ValueDecl *))
 ERROR(noncopyable_cannot_have_read_set_accessor,none,
-      "noncopyable %select{variable|subscript}0 cannot provide a read and set accessor",
+      "noncopyable %select{variable|subscript}0 cannot provide a read and set member",
```

Comment:
@kavon Can we remove `member` here completely and use `cannot provided both a read and set`?

Reply:
it will be like this: 
382b6131e14
i applied the change, i can revert it if it's not the best option <3

Reply:
As an aside, we could improve this message by:
* Displaying the descriptive declaration kind (`const ValueDecl *` +  `%kindonly0`) instead of a selection.
* Adding "both" for stress.
* Surrounding the accessor keywords with apostrophes.

Reply:
maybe i'm doing something in correct, but trying @AnthonyLatsis leave me with these new messages, i think they are not clear (or incorrect?)

```swift

public subscript(x x: Int) -> NonCopyableS { // expected-error {{noncopyable subscript cannot provide a read and set member}}
                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                                       /*suggested comment*/ -> noncopyable setter cannot provide both read and set `accessor`

```
and for variables, it would be: 



```swift
public var x: NonCopyableS { // expected-error {{noncopyable variable cannot provide a read and set member}}
                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                                   /*suggested comment*/ ->   noncopyable property cannot provide both read and set `accessor`




i find `subscript` one is less helpful ? i get that subscript is (setter/setting) by defination, but are we getting any value from replacing them with the `descriptiveDeclaration` ? 

i'm fine with adding `both` and `apostrophes ` btw

Reply:
A subscript can be read-only. You appear to be passing a setter instead of the subscript to the `diagnose` method.